pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,429 @@
+                a.show()
+                self.layout.addItem(a, y, vx)
+                shown_axes["y"][a.name] = a
+        return shown_axes
+
+    def _show_charts(self, charts=None):
+        """Shows all the selected charts."""
+        # SELECT CHARTS
+        if charts is None:
+            charts = self.charts
+        shown_charts = {}
+        for k, c in self.charts.items():
+            if k in charts:
+                c.show()
+                shown_charts[k] = c
+            else:
+                c.hide()
+        return shown_charts
+
+    def _connect_signals(self, top_level, chart):
+        """Connects all signals related to this widget for the given chart given the top level one."""
+        self._disconnect_all(chart)
+        top_vb = top_level.plotItem.vb
+        chart_vb = chart.plotItem.vb
+        signals = self._signalConnectionsByChart[chart.name]
+        scene = self.scene()
+        for axis_name in chart.axes:
+            # link axis to view
+            axis = self.axes[axis_name]
+            # connect view changes to axis
+            # set axis main view link if not assigned
+            if axis.linkedView() is None:
+                axis._linkedView = weakref.ref(chart_vb)
+            # FROM AxisItem.linkToView
+            # connect view changes to axis changes
+            if axis.orientation in ["right", "left"]:
+                signals["chart_vb.sigYRangeChanged"] = chart_vb.sigYRangeChanged.connect(
+                    axis.linkedViewChanged)
+            elif axis.orientation in ["top", "bottom"]:
+                signals["chart_vb.sigXRangeChanged"] = chart_vb.sigXRangeChanged.connect(
+                    axis.linkedViewChanged)
+            signals["chart_vb.sigResized"] = chart_vb.sigResized.connect(
+                axis.linkedViewChanged)
+            axis_view = axis.linkedView()
+            if chart_vb is not axis_view:
+                # FROM ViewBox.linkView
+                # connext axis's view changes to view since axis acts just like a proxy to it
+                if axis.orientation in ["top", "bottom"]:
+                    # connect axis main view changes to view
+                    chart_vb.state["linkedViews"][chart_vb.XAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    signals["axis_view.sigXRangeChanged"] = axis_view.sigXRangeChanged.connect(
+                        chart_vb.linkedXChanged)
+                    signals["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        chart_vb.linkedXChanged)
+                    # disable autorange on manual movements
+                    signals["axis_view.sigXRangeChangedManually"] = axis_view.sigXRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
```

Comment:
signal connected to a lambda function that references `self`

Reply:
One workaround I used for this is to put the important string onto the object, such as by doing the following:
```python
        # ...
        axis_view.setObjectName(axis_name)
        axis_view.sigXRangeChangedManuall(self._disableAxisFromNamedSender)
        # ...
    def _disableAxisFromNamedSendre(self):
        obj = self.sender()
        axis_name = obj.getObjectName()
        # ...
```

Reply:
added workaround